### ISSUE TITLE
xy and wh calculation

### ISSUE THREAD

**Issue by leoluopy** [deepcam-cn/yolov5-face#155]
i notice that ， while in loss calculation and inference decoding :
for x ,y and w,h calculation:  (yolo.py)
59:                y = y + torch.cat((x[i][:, :, :, :, 0:5].sigmoid(), torch.cat((x[i][:, :, :, :, 5:15], x[i][:, :, :, :, 15:15+self.nc].sigmoid()), 4)), 4)

61:                box_xy = (y[:, :, :, :, 0:2] * 2. - 0.5 + self.grid[i].to(x[i].device)) * self.stride[i] # xy
62:                box_wh = (y[:, :, :, :, 2:4] * 2) ** 2 * self.anchor_grid[i] # wh

why multiply *2 , and minus 0.5 in xy , actually , after sigmoid and add grid * stride . it's enough to cover the whole feat map.
dose it work better ? what's the principle ? 
@derronqi any idea ?

**Comment by leoluopy** [comment]
i guess the design for *2 -0.5 and *2 ) **2 is to get a larger backward optimization scope for xy and wh , is it right ? @derronqi
